### ISSUE TITLE
Sound wrongly played on editor focus when previous location contained an audio cue

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#143731]
Testing #143550

Turn on audio cues, for example for breakpoints or errors.

Have two editors side by side. Left editor has a line with a breakpoint, right one has no audio cues.
Focus on the left editor cursor navigate to the line with a breakpoint. Sound is played, great
Focus right editor. No sound -> great
Click on left editor on some line that does not have a breakpoint. Breakpoint sound is wrongly played 🐛

**Comment by hediet** [comment]
Nice catch!

The issue was that clicking into an unfocused editor would first focus the editor and only then update the position.
In my implementation of debounced observables, the observable would be defined immediately and only debounce following updates.

Thus, when focusing an editor and all observables are recreated, the audio cue was played immediately and then after the debounce it would potentially play again.

**Comment by isidorn** [comment]
Thanks for fixing 👏 
Adding to February milestone so I verify this tomorrow.
